### DISCUSSION TITLE
Shlink returns a 404 when visiting the base URL

### DISCUSSION THREAD

**Discussion by nothing2obvi** [shlinkio/shlink#2123] [Help wanted]
### Shlink version

latest

### PHP version

n/a

### How do you serve Shlink

Docker image

### Database engine

SQLite

### Database version

sqlite

### Summary

I can't get Shlink running, using either `DEFAULT_DOMAIN=shlink.mydomain.com` or `DEFAULT_DOMAIN=<my-ip>:<port>` with any combination of `IS_HTTPS_ENABLED=true` or `IS_HTTPS_ENABLED=false`. I'm able to successfully use a geolite license key. These are the only logs I get:

```
05/09/2024
11:31:21 PM
Initializing database if needed... Success!
05/09/2024
11:31:24 PM
Updating database... Success!
05/09/2024
11:31:24 PM
Generating proxies... Success!
05/09/2024
11:31:25 PM
Clearing entities cache... Success!
05/09/2024
11:31:32 PM
Downloading GeoLite2 db file... Success!
05/09/2024
11:31:35 PM
address=
tcp://127.0.0.1:6001
level=
debug
list of the plugins with RPC methods:=
["jobs","app","lock","resetter","informer"]
logger=
rpc
msg=
plugin was started
ts=
1715322695134521600
05/09/2024
11:31:35 PM
2024-05-10T06:31:35+0000	DEBUG	jobs        	initializing driver	{"pipeline": "shlink", "driver": "memory"}
05/09/2024
11:31:35 PM
2024-05-10T06:31:35+0000	DEBUG	jobs        	driver ready	{"pipeline": "shlink", "driver": "memory", "start": "2024-05-10T06:31:35+0000", "elapsed": 11}
05/09/2024
11:31:35 PM
driver=
memory
elapsed=
52.625µs
level=
debug
logger=
memory
msg=
pipeline was started
pipeline=
shlink
start=
2024-05-10 06:31:35.151001924 +0000 UTC
ts=
1715322695151005700
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:36 PM
2024-05-10T06:31:36+0000	DEBUG	jobs        	exited from jobs pipeline processor
05/09/2024
11:31:37 PM
[INFO] RoadRunner server started; version: 2023.3.12, buildtime: 2024-02-29T18:24:03+0000
05/09/2024
11:31:37 PM
[INFO] sdnotify: not notified
```

Navigating to the website using `<my-ip>:port` or `shlink.mydomain.com` gives a 404 error.

Any help would be appreciated. Thanks!

**Comment by acelaya** [comment]
That's the output of a perfectly working Shlink instance which is not receiving requests, so I imagine te problem is somewhere "before" Shlink, like a reverse proxy, CDN, etc.

If the 404 was served by Shlink, there would be an access log line printed, so you need to find out first who's causing it.

The `DEFAULT_DOMAIN` and `IS_HTTPS_ENABLED` env vars do not affect routing. Everything would still work regardless of their values. They are used to a) generate correct short URLs in API responses and b) determine if a request has been made to the default domain or a different one.

**Comment by nothing2obvi** [comment]
I could definitely be wrong but I think it might be served by Shlink because I get the Shlink favicon on the browser tab and the 404 looks like below, which is different from an error I'd get just on my browser.

![Screen Shot 2024-05-10 at 1 58 19 AM](https://github.com/shlinkio/shlink/assets/85228035/1d2d95e8-0378-4440-b1e8-9ba4345fc348)

I tried removing the network settings from the `docker-compose.yml` file but that didn't help either. I have no bind-mounts, currently, if that matters.

**Reply by acelaya** [reply]
Ah, yeah, that's in fact Shlink's 404 page, but that's actually the expected result when visiting Shlink's base URL 😅 

You can configure [special redirects](https://shlink.io/documentation/advanced/special-redirects/) for people visiting it, but other than that, all working as expected.

**Comment by nothing2obvi** [comment]
Thanks for your answer, but I'm sorry, I'm still not sure what to do. I tried adding the `DEFAULT_BASE_URL_REDIRECT` variable as `shlink.mydomain.com` and that only gave me something like `<my-ip>:<port>/shlink.mydomain.com`.

**Reply by acelaya** [reply]
First of all, it's important to understand what are your expectations, because it seems like there's some misconception on your behalf. 

What do you expect to see when visiting Shlink's base URL?

**Reply by nothing2obvi** [reply]
I expect to see some sort of webUI. Where would I find that? It sounds like I'm misunderstanding something.

**Reply by acelaya** [reply]
Shlink has been architected with an API-first approach in mind, and therefore, it does not provide a UI by itself. You need to use [shlink-web-client](https://shlink.io/documentation/shlink-web-client/) for that. The simplest way to do it is by using https://app.shlink.io and configuring your server/s there.

You can see some context on Shlink's architectural decisions [in the docs](https://shlink.io/documentation/advanced/shlink-architecture/).

**Reply by nothing2obvi** [reply]
Ohhhh, I see, there's a web client. I've got it now. Thanks for your help!
